swri_roscpp/include/swri_roscpp/subscriber_impl.h

```diff
@@ -68,34 +68,36 @@ class SubscriberImpl
     // timeout count.
     checkTimeout(now);
 
-    message_count_++;
-
-    if (!stamp.isZero()) {
-      ros::Duration latency = now - stamp;
-      if (message_count_ == 1) {
-        min_latency_ = latency;
-        max_latency_ = latency;
-        total_latency_ = latency;
-      } else {
-        min_latency_ = std::min(min_latency_, latency);
-        max_latency_ = std::max(max_latency_, latency);
-        total_latency_ += latency;
+    // Do not use this message for statistics if it is arriving out of order
+    if (stamp > last_header_stamp_) {
```

Comment:
I think this should be a `>=`

Reply:
Yeah, seems like that would result in better behavior.